impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -286,7 +286,7 @@ - (void)flutterPrepareForPresent:(nonnull id<MTLCommandBuffer>)commandBuffer;
     // If the threads have been merged, or there is a pending frame capture,
     // then block on cmd buffer scheduling to ensure that the
     // transaction/capture work correctly.
```

Comment:
The above comment isn't adding much and rather than making it even longer to cover the new case, we could probably just yank out a local and the code would do the exact same amount of documenting as the comment currently does:
```ObjC
BOOL threads_merged = [[NSThread currentThread] isMainThread];
```
Or maybe extract the whole blobby conditional to a bool named `transactions_enabled` or something.

Reply:
Maybe I can move this to the surface_frame doc comment?

Reply:
Sounds good.